File: src/main/java/com/kingmang/lazurite/exceptions/VariableDoesNotExistsException.java
package com.kingmang.lazurite.exceptions;

import lombok.Getter;

@Getter
public final class VariableDoesNotExistsException extends LzrException {

    private final String variable;

    public VariableDoesNotExistsException(String variable) {
        super("VariableDoesNotExistsException", "Variable: \"" + variable + "\" does not exists");
        this.variable = variable;
    }

}

File: src/main/java/com/kingmang/lazurite/exceptions/VariableDoesNotExistsException.kt
package com.kingmang.lazurite.exceptions

class VariableDoesNotExistsException : LzrException {
    val variable: String

    constructor(variable: String) : super("VariableDoesNotExistsException", "Variable \"$variable\" does not exists") {
        this.variable = variable
    }

    constructor(variable: String, message: String) : super("VariableDoesNotExistsException", "Variable \"$variable\" does not exists. $message") {
        this.variable = variable
    }
}